mono/eglib/gmodule-win32.c

```diff
@@ -189,3 +192,155 @@ g_module_build_path (const gchar *directory, const gchar *module_name)
 	}
 	return g_strdup_printf ("%s%s" LIBSUFFIX, lib_prefix, module_name); 
 }
+
+// This is not about GModule but is still a close fit.
+// This is not named "g_" but that should be ok.
+// g_free the result
+// No MAX_PATH limit.
+//
+// Prefer mono_get_module_filename over mono_get_module_filename_ex and mono_get_module_basename.
+// Prefer not-ex, not-base.
+//
+gboolean
+mono_get_module_filename (gpointer mod, gunichar2** pstr, guint32* plength)
+{
+	gunichar2* str = NULL;
+	guint32 capacity = 32; // tunable
```

Comment:
Tune up for final commit? Note that it double first, so this is 64 note 32, as it might appear.

Reply:
I'd definitely start it higher since ideally we only have a single api call in most cases. I just used 260 for my PR since, while arbitrary, it preserves existing behavior as best as possible.

Reply:
That is usually a waste of memory though.

Reply:
A lot of code bases including CLR will "inline" (i.e. stack) allocate something reasonable like 64 and then only use heap for larger. Nicely packaged into a templatized class.

Reply:
Yeah it is, but I think that's probably preferable to regular repeated api calls. If it was actually a serious concern it'd be worth tuning the value, but we can probably just put it at that and call it good. The biggest concern is just that it doesn't truncate longer paths.

Reply:
There's no perfect solution here and several flawed ones.
MAX_PATH also takes away coverage. Half the code becomes essentially dead and untested.
You kinda want it to be slow, sometimes.
Or maybe force it under repeated testing, i.e. test short and long paths in CI. Hopefully not taking down the rest of the system w/ long paths.

Reply:
Yeah, I think the best solution here is to add a test that makes use of a longer path. We know it sometimes comes up in integration CI at least, so the code is exercised eventually, but ideally it would be further upstream.